utils/eps_to_csv_converter.py

```diff
@@ -0,0 +1,428 @@
+    """
+    client_id = os.environ.get("EPS_CLIENT_ID")
+    host = os.environ.get("EPS_HOST")
+    service_account = os.environ.get("SERVICE_ACCOUNT")
+
+    missing_vars = []
+    if client_id is None:
+        missing_vars.append("EPS_CLIENT_ID")
+    if host is None:
+        missing_vars.append("EPS_HOST")
+    if service_account is None:
+        missing_vars.append("SERVICE_ACCOUNT")
+
+    if missing_vars:
+        # Raise ValueError for configuration issues
+        raise ValueError(
+            f"Missing required environment variables: {', '.join(missing_vars)}"
+        )
+
+    logger.info("Successfully retrieved parameters from environment variables.")
+    return EPSParameters(
+        client_id=client_id, host=host, service_account=service_account
+    )
+
+
+# --- IAP Request Function ---
+def make_iap_request(
+        url: str, client_id: str, service_account: str, method: str = "GET", **kwargs
+) -> Dict:
+    """Makes a request to an application protected by Identity-Aware Proxy.
+
+    Args:
+      method: The HTTP method to make the request
+      service_account: The Target Service Account Email to be impersonated
+      url: The Identity-Aware Proxy-protected URL to fetch.
+      client_id: The client ID used by Identity-Aware Proxy.
+      **kwargs: Any parameters for requests.request.
+
+    Returns:
+      The parsed JSON response body as a dictionary.
+
+    Raises:
+        NotImplementedError: If method != GET (other methods can be implemented later)
+        ValueError: If the response is not valid JSON.
+        Exception: If the request fails or returns a non-200 status.
+    """
+    if method.upper() != "GET":  # Added method handling example
+        logger.error(f"HTTP method '{method}' not currently supported.")
+        raise NotImplementedError(f"HTTP method '{method}' not implemented.")
+
+    if "timeout" not in kwargs:
+        kwargs["timeout"] = 90
+
+    try:
+        target_service_account_email = service_account
+        audience = client_id
+        logger.debug(
+            f"Generating ID token for SA '{target_service_account_email}' with audience '{audience}'"
+        )
+        client = iam_credentials_v1.IAMCredentialsClient()
+        name = f"projects/-/serviceAccounts/{target_service_account_email}"
+        id_token_response = client.generate_id_token(
+            name=name, audience=audience, include_email=True
+        )
+        id_token_jwt = id_token_response.token
+        logger.debug("Creating authorized session with generated ID token.")
+        iap_creds = google.oauth2.credentials.Credentials(id_token_jwt)
+        authed_session = AuthorizedSession(iap_creds)
+
+        logger.info(f"Making {method} request to IAP URL: {url}")
+
+        resp = authed_session.request(method, url, **kwargs)
+        logger.info(f"Received status code: {resp.status_code} from {url}")
+        resp.raise_for_status()  # Raises HTTPError for bad responses (4xx or 5xx)
```

Comment:
Note: it might make sense to wrap this in a `try/except` to catch the `HTTPError` rather than do a full traceback without a message

```
try:
    resp.raise_for_status()
except HTTPError as e:
    logger.error(f'Got response status code: {resp.status_code()')
    sys.exit(1)
```